misc/toml-parser/src/main/java/io/ballerina/toml/internal/parser/TomlParser.java

```diff
@@ -445,35 +470,118 @@ private STNode parseBoolean() {
+     *
+     * @return Multiline String Literal Node.
+     */
+    private STNode parseMultilineLiteralStringValue() {
+        STNode startingDoubleQuote = parseMultipleSingleQuoteToken(ParserRuleContext.MULTILINE_LITERAL_STRING_START);
+        STNode content = parseStringContent(ParserRuleContext.MULTILINE_LITERAL_STRING_CONTENT);
+        STNode endingDoubleQuote = parseMultipleSingleQuoteToken(ParserRuleContext.MULTILINE_LITERAL_STRING_END);
+        return STNodeFactory.createLiteralStringLiteralNode(startingDoubleQuote, content, endingDoubleQuote);
+    }
+
+    /**
+     * Parse Single quote token.
+     *
+     * @return Single quote token
+     */
+    private STNode parseSingleQuoteToken(ParserRuleContext ctx) {
+        STToken token = peek();
+        if (token.kind == SINGLE_QUOTE_TOKEN) {
+            return consume();
+        } else {
+            recover(token, ctx);
+            return parseSingleQuoteToken(ctx);
+        }
+    }
+
+    /**
+     * Parse Single quote token.
+     *
+     * @return Single quote token
+     */
+    private STNode parseMultipleSingleQuoteToken(ParserRuleContext ctx) {
+        STToken token = peek();
+        if (token.kind == TRIPLE_SINGLE_QUOTE_TOKEN) {
+            return consume();
+        } else {
+            recover(token, ctx);
+            return parseMultipleSingleQuoteToken(ctx);
+        }
+    }
+
     /**
      * Parse Double quote token.
      *
      * @return Double quote token
      */
     private STNode parseDoubleQuoteToken(ParserRuleContext ctx) {
         STToken token = peek();
-        if (token.kind == SyntaxKind.DOUBLE_QUOTE_TOKEN || token.kind == SyntaxKind.TRIPLE_DOUBLE_QUOTE_TOKEN
-                || token.kind == SINGLE_QUOTE_TOKEN) {
+        if (token.kind == SyntaxKind.DOUBLE_QUOTE_TOKEN) {
             return consume();
         } else {
             recover(token, ctx);
             return parseDoubleQuoteToken(ctx);
         }
     }
 
-    private STNode parseStringContent() {
+    /**
+     * Parse Triple Double quote token.
+     *
+     * @return Triple Double quote token
+     */
+    private STNode parseTripleDoubleQuoteToken(ParserRuleContext ctx) {
+        STToken token = peek();
+        if (token.kind == SyntaxKind.TRIPLE_DOUBLE_QUOTE_TOKEN) {
+            return consume();
+        } else {
+            recover(token, ctx);
+            return parseTripleDoubleQuoteToken(ctx);
+        }
+    }
+
+    private STNode parseStringContent(ParserRuleContext ctx) {
         STToken nextToken = peek();
         if (nextToken.kind == SyntaxKind.IDENTIFIER_LITERAL) {
             return consume();
-        } else {
-            recover(nextToken, ParserRuleContext.STRING_CONTENT);
-            return parseStringContent();
         }
+        if (nextToken.kind == SyntaxKind.DOUBLE_QUOTE_TOKEN || nextToken.kind == SINGLE_QUOTE_TOKEN ||
+                nextToken.kind == TRIPLE_SINGLE_QUOTE_TOKEN || nextToken.kind == TRIPLE_DOUBLE_QUOTE_TOKEN) {
```

Comment:
We have a small problem here. We call parseStringContent in 4 places and depending on the context we need only one kind to be checked here. Here we are heavily depending on the lexer logic. i.e we assume SINGLE_QUOTE_TOKEN is always followed by SINGLE_QUOTE_TOKEN when the string content is empty. We can make parsing independent of lexer logic by having 4 separate methods to parse each.